src/grt/src/GlobalRouter.cpp

```diff
@@ -787,6 +782,27 @@ bool GlobalRouter::makeFastrouteNet(Net* net)
   return false;
 }
 
+void GlobalRouter::getNetLayerRange(Net* net,
+                                    int& min_layer,
+                                    int& max_layer)
+{
+  int port_min_layer = std::numeric_limits<int>::max();
+  for (const Pin& pin : net->getPins()) {
+    if (pin.isPort()) {
+      port_min_layer = std::min(port_min_layer, pin.getTopLayer());
+    }
+  }
+
+  bool is_non_leaf_clock = isNonLeafClock(net->getDbNet());
```

Comment:
It looks like isNonLeafClock looks for register but not for pads.  I think in bp_single the connection is a clock net to a pad.

Reply:
I've tried to add it in my last commit, but not sure if it is the correct way.